storage/storage_discoverHint.go

```diff
@@ -0,0 +1,72 @@
+package storage
+
+import (
+	"github.com/bmc-toolbox/dora/filter"
+	"github.com/bmc-toolbox/dora/model"
+	"github.com/jinzhu/gorm"
+)
+
+// NewDiscoverHintStorage initializes the storage
+func NewDiscoverHintStorage(db *gorm.DB) *DiscoverHintStorage {
+	return &DiscoverHintStorage{db}
+}
+
+// DiscoverHintStorage stores all DiscoverHints
+type DiscoverHintStorage struct {
+	db *gorm.DB
+}
+
+// Count gets DiscoverHints count based on the filter
+func (s DiscoverHintStorage) Count(filters *filter.Filters) (count int, err error) {
+	q, err := filters.BuildQuery(model.DiscoverHint{}, s.db)
+	if err != nil {
+		return count, err
+	}
+
+	err = q.Model(&model.DiscoverHint{}).Count(&count).Error
+	return count, err
+}
+
+// GetAll of the DiscoverHints
+func (s DiscoverHintStorage) GetAll(offset string, limit string) (count int, ips []model.DiscoverHint, err error) {
+	if offset != "" && limit != "" {
+		if err = s.db.Limit(limit).Offset(offset).Order("ip").Find(&ips).Error; err != nil {
+			return count, ips, err
+		}
+		s.db.Model(&model.DiscoverHint{}).Order("ip").Count(&count)
+	} else {
+		if err = s.db.Order("ip").Find(&ips).Error; err != nil {
+			return count, ips, err
+		}
+	}
+	return count, ips, err
+}
+
+// GetAllByFilters get all Discover Hints based on the filter
+func (s DiscoverHintStorage) GetAllByFilters(offset string, limit string, filters *filter.Filters) (count int, ips []model.DiscoverHint, err error) {
+	q, err := filters.BuildQuery(model.DiscoverHint{}, s.db)
+	if err != nil {
+		return count, ips, err
+	}
+
+	if offset != "" && limit != "" {
```

Comment:
Same comment as earlier. Is it valid to provide just `offset` or just `limit` ?

Reply:
Fixed